packages/botonic-plugin-flow-builder/src/action/index.tsx

```diff
@@ -103,3 +117,17 @@ async function getFallbackNode(cmsApi: FlowBuilderApi, request: ActionRequest) {
   await trackEvent(request, event)
   return fallbackNode
 }
+
+function replaceVariables(text: string, extraData?: Record<string, any>) {
+  const matches = text.match(VARIABLE_REGEX)
+
+  if (matches && extraData) {
+    matches.forEach(match => {
+      const variable = match.slice(1, -1)
+      const value = extraData[variable] ? extraData[variable] : match
+      text = text.replace(match, value)
+    })
+  }
+
+  return text
+}
```

Comment:
maybe this can be moved inside the `FlowText` class

Reply:
> maybe this can be moved inside the `FlowText` class

yup, or create a function utils for different components. If it is only needed for this component, then makes sense to have it within this class out of the scope of the class